src/sources/groups.js

```diff
@@ -255,6 +258,13 @@ class GroupNarrative extends Narrative {
 }
 
 
+class GroupOverview extends Overview {
+  get baseName() {
+    return "group-overview";
+  }
+}
```

Comment:
https://github.com/nextstrain/nextstrain.org/labels/request%20for%20comments

I initially started with 0010a642285089dbc3befc79a82b1d405dd6c639...4f5452ce84d4a5e32714f7a617997894d20204d4 which gets the URL by `GroupSource.urlFor("group-overview.md")` and uses the group's [`authz.tags.Type.Source`](https://github.com/nextstrain/nextstrain.org/blob/da26d59de198471b8948532ec3e52f2c2d780456/src/authz/tags.js#L17). Then I switched to 46990fdee31be7137ecb3f451fb61abbd23ff0c4 which object-ifys `group-overview.md`, comparable to datasets/narratives. These are 2 very different approaches. Is the second one overkill?

Reply:
Ah, yeah. I think the second is probably overkill since this is a single, system-defined file as opposed to arbitrary user-defined sets of files.

Reply:
One consideration would be if we need write ACLs on the overview to be separate from write ACLs on the source itself, but I don't think we do. As described [in the group roles docs I wrote](https://docs.nextstrain.org/en/latest/learn/groups/index.html#roles), I think editing the overview should be purview of the owners role.

Reply:
Ok, I'll revert back then. But I did learn much more when making all the classes to get the second working, so that exercise didn't go to waste!

Reply:
I'm thinking about authz again and how it relates to groups as part of implementing the API for group membership management. In that work, I need to control the ability to modify the group itself (e.g. add/remove members) separately from the ability to modify the _contents_ of the group (e.g. the group's Source, with its datasets and narratives). I'm (so far) opting to add `authz.tags.Type.Group` and run authz checks for the membership management endpoints against the `Group` instead of the `GroupSource`. This had me thinking about our brief discussion here.

I realized that how I'm extending the authz system is applicable to modifying the group overview and logo, and perhaps is what those endpoints here should be authz'd using.

In re-reading my commentary here, I also realized I made a mistake. I said:

> One consideration would be if we need write ACLs on the overview to be separate from write ACLs on the source itself, but I don't think we do. As described [in the group roles docs I wrote](https://docs.nextstrain.org/en/latest/learn/groups/index.html#roles), I think editing the overview should be purview of the owners role.

But the ACLs on the GroupSource itself allow not just owners the Write action, but also editors. This is contrary to what I think should be the case!

Reply:
I see, yeah the current implementation allows editors to edit overview when it should be just owners.

> add `authz.tags.Type.Group` and run authz checks for the membership management endpoints against the `Group` instead of the `GroupSource`

The difference between `Group` vs. `GroupSource` isn't entirely clear to me. Is the `Group` also available in `req.context` like the `GroupSource` currently is through `req.context.source`?

https://github.com/nextstrain/nextstrain.org/blob/25bccef5b004c3e4965c8ba6c2af3c9d0cf0fa15/src/endpoints/sources.js#L319-L323

Reply:
> the ACLs on the GroupSource itself allow not just owners the Write action, but also editors.

Is this true? Based on the snippet below, I don't think any role currently has Write access to the GroupSource itself:

https://github.com/nextstrain/nextstrain.org/blob/0975bc0650f13c8b0a7abd57f9d8c7e5cd0e5684/src/sources/groups.js#L279-L284

From your statement I'd expect it to be:

```js
{tag: Type.Source, role: viewers, allow: [Read]}, 
{tag: Type.Source, role: editors, allow: [Read, Write]}, 
{tag: Type.Source, role: owners,  allow: [Read, Write]}, 
```

Reply:
> The difference between `Group` vs. `GroupSource` isn't entirely clear to me.

My mental model distinguishes between the concepts of groups and sources like so:

Sources are abstract containers for datasets and narratives and provide the interface for shared code to work across the various concrete ways we store/access datasets and narratives (core, staging, community, "fetch", groups, etc). Sources are a concept mostly internal to the codebase, i.e. not exposed to users.

Groups have members, roles, a visibility, overview/description, a logo, etc. and also contain datasets and narratives. So every group has a source, but not all sources have a group. Groups are an externally-visible concept, i.e. exposed to users.

See also the [glossary](https://docs.nextstrain.org/projects/nextstrain-dot-org/en/latest/glossary.html#term-Source) and [previous commentary](https://github.com/nextstrain/nextstrain.org/issues/76#issuecomment-574303470).

This said, there are likely some places in the codebase where these two concepts are currently a bit blurred together. However, I think we should be further distinguishing them over time. (And this is what I'm doing in my work on implementing the API for group membership management.)

> Is the `Group` also available in `req.context` like the `GroupSource` currently is through `req.context.source`?

Not currently, as it's not needed, but it could be.

> Is this true? Based on the snippet below, I don't think any role currently has Write access to the GroupSource itself:

You're right! :man_facepalming: I confused myself on my earlier re-read and thought I'd made a mistake I hadn't, thus making a new mistake instead. :upside_down_face:

Reply:
Ok! Sounds like a new `authz.tags.Type.Group` will be ideal here instead of 2adabdb426743f2c4b254ceb2cbc7bcbe1e9ec6b. If you already have work in progress for that, I don't mind waiting until that's merged to use it here.

Reply:
Ok. See https://github.com/nextstrain/nextstrain.org/pull/581/commits/b54b35ac4309b9a0d8c0f792b8fb8508d0793c72.

Reply:
Planning to rebase onto b54b35ac4309b9a0d8c0f792b8fb8508d0793c72 and move all the changes in `src/endpoints/sources.js` to a new `src/endpoints.groups.js`, to be consistent with #581.

Reply:
Maybe base off 29449eb instead? Contains work I'd forgotten to push up yesterday.

Reply:
You can see an example of how it's used in 9431522e9c099c0b970a25f8b179e7da887e81f9.

Reply:
Done. Added some new thoughts after the rebase: https://github.com/nextstrain/nextstrain.org/pull/570#pullrequestreview-1087578047